scripts/smoke-test-droid.ps1

```diff
@@ -0,0 +1,112 @@
+# GITHUB_WORKSPACE is the root folder where the project is stored.
+Write-Output "#################################################"
+Write-Output "#   ANDROID                                     #"
+Write-Output "#            VALIDATOR                          #"
+Write-Output "#                       SCRIPT                  #"
+Write-Output "#################################################"
+
+Set-Variable -Name "ApkPath" -Value "samples/artifacts/builds/Android"
+Set-Variable -Name "ApkFileName" -Value "IL2CPP_Player.apk"
+Set-Variable -Name "ActivityName" -Value "io.sentry.samples.unityofbugs"
+Set-Variable -Name "TestActivityName" -Value "io.sentry.samples.unityofbugs/com.unity3d.player.UnityPlayerActivity"
+
+# Filter device List
+$RawAdbDeviceList = adb devices
+$deviceList = @()
+foreach ($device in $RawAdbDeviceList)
+{
+    if ($device.EndsWith("device"))
+    {
+        $deviceList += $device.Replace("device", '').Trim()
+    }
+}
+$deviceCount = $deviceList.Count
+
+if ($deviceCount -eq 0)
+{
+    Throw "It seems like no devices were found $RawAdbDeviceList"
+}
+else
+{
+    Write-Output "Found $deviceCount devices: $deviceList"
+}
+
+# Check if APK was built.
+if (Test-Path -Path "$ApkPath/$ApkFileName" ) 
+{
+    Write-Output "Found $ApkPath/$ApkFileName"
+}
+else
+{
+    Throw "Expected APK on $ApkPath/$ApkFileName but it was not found."
+}
+
+# Test
+foreach ($device in $deviceList)
+{
+    $deviceSdk = adb shell getprop ro.build.version.sdk 
+    $deviceApi = adb shell getprop ro.build.version.release 
+    Write-Output ""
+    Write-Output "Checking device $device with SDK $deviceSdk and API $deviceApi"
+    Write-Output ""
+
+    adb -s $device uninstall $ActivityName
+    $stdout = (adb -s $device install -r $ApkPath/$ApkFileName)
+    if($stdout -notcontains "Success")
+    {
+        Throw "Failed to Install APK: $stdout."
+    }
+    $checkStarted = 'False'
+
+    Write-Output "Clearing logcat from $device."
+    adb -s $device logcat -c
+
+    Write-Output "Starting Test..."
+
+    adb -s $device shell am start -n $TestActivityName -e test smoke
+
+    for ($i = 30; $i -gt 0; $i--) {
```

Comment:
What about something like this:
```
$timeout = 30
while($timeout -gt 0){
    $smokeTestId = (adb -s $device shell ps)
    $smokeTestId = $smokeTestId | select-string $ActivityName

    If ($smokeTestId -eq $null)
    {
        break
    }

    Write-Output "Waiting Process on $device to complete, waiting $timeout seconds"
    Start-Sleep -Seconds 1
    $timeout--
}
```
With the `checkStarted` you skip ahead in the for without waiting so the timeout of 30 is not really 30s?

Reply:
Some Android versions seem to only exit the adb callback for `am start` when the app is open, while others don't.
So checkStarted is there to inform if the app was started or not.

If the app was not initialized, smokeTestId is going to be set to null, and checkStarted will avoid going out of the loop since the app wasn't initialized.

Once it detects the app was initialized, it'll wait for smokeTestId  to be come null.

About smokeTestId, it'll get the list of active processes, and doing the `$smokeTestId | select-string $ActivityName`
 will return null if it doesn't find `io.sentry.samples.unityofbugs` otherwise, it'll return ugh, `io.sentry.samples.unityofbugs` 😅.

It's a silly way of checking if the process is running, there're better ways of doing that, but those methods may run on version X and not on Y where the PS one seems to work on any Android version.